tests/testsuite.src/syn_misc.at

```diff
@@ -8592,3 +8592,93 @@ prog.cob:9: warning: DISPLAY statement in DEFAULT SECTION is not implemented
+
+
+AT_SETUP([PICTURE L])
+AT_KEYWORDS([extensions gcos picture-l])
+
+AT_DATA([prog_extraneous_depending.cob], [
+       IDENTIFICATION   DIVISION.
+       PROGRAM-ID.      prog_extraneous_depending.
+       DATA             DIVISION.
+       WORKING-STORAGE  SECTION.
+       1 L		COMP-1.
+       1 F-DAT		PICTURE X(10) DEPENDING ON L.
+       PROCEDURE        DIVISION.
+           MOVE SPACES TO F-DAT
+           STOP RUN.
+])
+AT_CHECK([$COMPILE_ONLY prog_extraneous_depending.cob], [1], [],
+[prog_extraneous_depending.cob:7: error: DEPENDING clause must either come with an OCCURS clause or a PICTURE string with 'L' character
+])
+
+AT_DATA([prog_missing_depending.cob], [
+       IDENTIFICATION   DIVISION.
+       PROGRAM-ID.      prog_missing_depending.
+       DATA             DIVISION.
+       WORKING-STORAGE  SECTION.
+       1 F-DAT		PICTURE LX(10).
+       PROCEDURE        DIVISION.
+           MOVE SPACES TO F-DAT
+           STOP RUN.
+])
+
+AT_CHECK([$COMPILE_ONLY prog_missing_depending.cob], [1], [],
+[prog_missing_depending.cob:6: error: missing DEPENDING clause for PICTURE string with 'L' character
+])
+
+AT_DATA([prog_value.cob], [
+       IDENTIFICATION   DIVISION.
+       PROGRAM-ID.      prog_value.
+       DATA             DIVISION.
+       WORKING-STORAGE  SECTION.
+       1 L		COMP-1 VALUE 10.
+       1 F-DAT		PICTURE LX(10) DEPENDING ON L
+                        VALUE "0123456789".
+       PROCEDURE        DIVISION.
+           STOP RUN.
+])
+AT_CHECK([$COMPILE_ONLY prog_value.cob], [1], [],
+[prog_value.cob:7: error: PICTURE L and VALUE are mutually exclusive
+])
+
+AT_DATA([prog_errs.cob], [
+       IDENTIFICATION   DIVISION.
+       PROGRAM-ID.      prog_errs.
+       DATA             DIVISION.
+       WORKING-STORAGE  SECTION.
+       1 W-LGS.
+         2 LGS		PICTURE X(10).
+       1 W-DATA.
+         2 L-DAT	PICTURE LX(10) DEPENDING ON LGS.
+	 2 L-ERR1       PICTURE L9 DEPENDING ON LGS.
+	 2 L-ERR2       PICTURE XXXXLA(2) DEPENDING ON LGS.
+	 2 L-ERR3       PICTURE LX(1) DEPENDING ON LGS JUSTIFIED.
+       PROCEDURE        DIVISION.
+           STOP RUN.
+])
+
+AT_CHECK([$COMPILE_ONLY prog_errs.cob], [1], [],
+[prog_errs.cob:10: error: only USAGE DISPLAY may specify a variable-length PICTURE string
+prog_errs.cob:11: error: L must be at start of PICTURE string
+prog_errs.cob:11: error: L cannot follow A or X
+prog_errs.cob:11: error: only USAGE DISPLAY may specify a variable-length PICTURE string
+prog_errs.cob:12: error: 'L-ERR3' cannot have JUSTIFIED RIGHT clause
+])
+
+AT_DATA([prog.cob], [
+       IDENTIFICATION   DIVISION.
+       PROGRAM-ID.      prog.
+       DATA             DIVISION.
+       WORKING-STORAGE  SECTION.
+       1 LGS		PICTURE X(10).
+       1 L-DAT		PICTURE LX(10) DEPENDING ON LGS.
+       PROCEDURE        DIVISION.
+           STOP RUN.
+])
+
+AT_CHECK([$COMPILE_ONLY -fpicture-l=warning prog.cob], [0], [],
+[prog.cob:7: warning: PICTURE string with 'L' character used
+])
+
+AT_CLEANUP
```

Comment:
Test missing: PIC L  may not have VALUE clause (and other stuff)

Reply:
Actually the message might not be clear enough: it's tested with `prog_value.cob` a few lines above that. The error is
```
prog_value.cob:7: error: PICTURE L and VALUE are mutually exclusive
```

Reply:
That's fine. I _thought_ this and other added syntax rules were missing from the tests, just give it a quick check and when you think you have those covered then that's fine.

Reply:
I checked the manual and found `BLANK WHEN ZERO` was (obviously) not allows for `PIC L` fields. This was already rejected but I clarified the message.
I found no other clauses that explicitly conflicted with such fields.
Just tell me if you think of now other clause, and if that can go upstream.